### PR TITLE
Resolved hang issue for CR-1131262

### PR DESCRIPTION
**Pull request by subeeshm-xilinx**
Resolved the unwanted creation on mmap files for non-p2p buffers 
This is an issue which was caught as part of the CR-1131262
There is no workaround for the solution
Low Risk
We had ran the P2P testcases and also the testcase provided for CR-1131262 manually. Also ran the canary suite
No impact on Documentation

### CODE REVIEW

**Review by keryell — APPROVED** [review]
Thanks!

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -1480,7 +1480,7 @@ uint32_t HwEmShim::getAddressSpace (uint32_t topology)
     uint64_t finalSize = size+(2*paddingFactor*size);
     mAddrMap[finalValidAddress] = finalSize;
     bool ack = false;
-    if(sock)
+    if (sock && (boFlags & XCL_BO_FLAGS_P2P)) // bypassed the xclAllocDeviceBuffer RPC call for Non-P2P
```
[the remark this answers is not in the document]
Comment (hemantk-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:1483]: Hi @venkatp-xilinx This call is mainly needed to create mmap file handle for P2P buffers for non p2p buffers this is not needed as of now 
Ideally we can still have this call for diagnostics purpose but not needed as of now

**Review by hemantk-xilinx — APPROVED** [review]

**Review by venkatp-xilinx — APPROVED** [review]

**Comment by subeeshm-xilinx** [comment]
Can anyone merge this pull request?

**Comment by chvamshi-xilinx** [comment]
Please rebase your branch as this PR is very old. It should pass the build with latest changes aswell.

**Pull request merged**